hw/dv/sv/sim_sram/index.md

```diff
@@ -0,0 +1,80 @@
+In DV, this is envisioned to be used for the following purposes:
+- write the status of the SW test
+- use as a SW logging device
+- write the output of an operation
+- signal an event
+
+These usecases apply to Verilator as well.
+However, at this time, the Verilator based simulations rely on the on-device UART for logging.
+
+This module must be instantiated before any XBAR / bus fabric element in the design which can return an error response when it sees an access to an invalid address range.
+Ideally, it should be placed as close to the CPU (or equivalent) as possible to reduce the simuation time.
+
+## Customizations
+
+`sim_sram` exposes the following parameters:
+
+- `InstantiateSram` (default: 0):
+
+  Controls whether to instantiate the SRAM or not.
+  The most typical (and a very much recommended) operating mode is for the SW test to only write data to this SRAM, to make it portable across various simulation platforms (DV, Verilator, FPGA etc.).
+  The testbench can simply probe the sinked output of the socket to monitor writes to specific addresses within this range.
+  So, in reality the actual SRAM instance is not even needed.
+  However, there may be some future usecases for instantiating it.
+
+- `BusAddrWidth` (default: 32):
+
+  This should be the same as the address width of the TL interface.
+
+- `SimSramWidth` (default: 32):
+
+  This should be the same as the data width of the TL interface.
+
+- `SimSramDepth` (default: 8):
+
+  Depth of the SRAM in bus words.
+
+- `ErrOnRead` (default: 1):
+
+  If the SW test reads from the SRAM, throw an error (see `InstantiateSram` above for more details).
+
+In addition, the module instantiates the `sim_sram_if` interface to allow the testbench to control the `start_addr` of the SRAM (defaults to 0) at run-time by hierarchically referencing it:
+```systemverilog
+  // In top level testbench
+  assign tb.foo.bar.u_sim_sram.u_sim_sram_if.start_addr = 32'h3000_0000;
+```
+This interface also exposes `tl_i`, which is connected to the sinked TL output of the socket (qualified TL access made to the SRAM).
+The handle to this interface can be supplied to the UVM side via `uvm_config_db` for fine-grained control and monitoring.
+
+## Usage
+
+This module needs to be instantiated on an existing outbound TL connection, possibly deep in an existing design hierarchy while also abiding by these:
+- design sources must not depend on simulation components
+- design sources must not directly instantiate simulation components, even with `ifdefs`
+- synthesis of the design is not disturbed whatsoever
+- we must be able to run simulations with Verilator (which prevents us from using forces for example)
+
+One way of achieving this is by disconnecting an outbound TL interface in the design source where `sim_sram` needs to be inserted.
+The disconnection must be made ONLY if one either `VERILATOR` or `UVM` macro is defined.
```

Comment:
Wouldn't it be better to have an explicit "MAGICMEM" define? Otherwise, you can't simulate the chip (without magical additions) under Verilator, or with UVM.

Reply:
Hmm, on second thoughts, only Verilator needs the disconnection. We can just use forces in DV.